samza-core/src/main/java/org/apache/samza/clustermanager/container/placement/ContainerPlacementMessageObjectMapper.java

```diff
@@ -0,0 +1,119 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.clustermanager.container.placement;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import org.apache.samza.container.placement.ContainerPlacementMessage;
+import org.apache.samza.container.placement.ContainerPlacementRequestMessage;
+import org.apache.samza.container.placement.ContainerPlacementResponseMessage;
+import org.codehaus.jackson.JsonGenerator;
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.JsonParser;
+import org.codehaus.jackson.ObjectCodec;
+import org.codehaus.jackson.Version;
+import org.codehaus.jackson.map.DeserializationContext;
+import org.codehaus.jackson.map.JsonDeserializer;
+import org.codehaus.jackson.map.JsonSerializer;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.codehaus.jackson.map.SerializerProvider;
+import org.codehaus.jackson.map.jsontype.NamedType;
+import org.codehaus.jackson.map.module.SimpleModule;
+
+/**
+ * Object mapper for serializing and deserializing Container placement messages
+ */
+public class ContainerPlacementMessageObjectMapper {
+
+  private ContainerPlacementMessageObjectMapper() {
+  }
+
+  static ObjectMapper getObjectMapper() {
+    ObjectMapper objectMapper = new ObjectMapper();
+    SimpleModule module = new SimpleModule("ContainerPlacementModule", new Version(1, 0, 0, ""));
+    module.addSerializer(ContainerPlacementMessage.class,
+        new ContainerPlacementMessageObjectMapper.ContainerPlacementMessageSerializer());
+    module.addDeserializer(ContainerPlacementMessage.class,
+        new ContainerPlacementMessageObjectMapper.ContainerPlacementMessageDeserializer());
+    objectMapper.registerModule(module);
+    objectMapper.registerSubtypes(new NamedType(ContainerPlacementResponseMessage.class));
+    objectMapper.registerSubtypes(new NamedType(ContainerPlacementRequestMessage.class));
+    return objectMapper;
+  }
+
+  static class ContainerPlacementMessageSerializer extends JsonSerializer<ContainerPlacementMessage> {
+    @Override
+    public void serialize(ContainerPlacementMessage value, JsonGenerator jsonGenerator, SerializerProvider provider)
+        throws IOException {
+      Map<String, Object> containerPlacementMessageMap = new HashMap<String, Object>();
+      if (value instanceof ContainerPlacementRequestMessage) {
+        containerPlacementMessageMap.put("subType", ContainerPlacementRequestMessage.class.getSimpleName());
+      } else if (value instanceof ContainerPlacementResponseMessage) {
+        containerPlacementMessageMap.put("subType", ContainerPlacementResponseMessage.class.getSimpleName());
+        containerPlacementMessageMap.put("responseMessage",
+            ((ContainerPlacementResponseMessage) value).getResponseMessage());
+      }
```

Comment:
Would you want to add an else check for unexpected subtypes of `ContainerPlacementMessage`? Also may apply to the `deserialize` method below.